test/util/client/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,448 @@
+	// Create the template
+	template, err := t.tc.Templates("openshift").Get(
+		tpl, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Instantiate the template
+	_, err = t.tc.TemplateInstances(t.namespace).Create(
+		&templatev1.TemplateInstance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: t.namespace,
+			},
+			Spec: templatev1.TemplateInstanceSpec{
+				Template: *template,
+			},
+		})
+	if err != nil {
+		return err
+	}
+
+	// Return after waiting for instance to complete
+	return wait.PollImmediate(2*time.Second, 10*time.Minute, t.TemplateInstanceIsReady)
+}
+
+func (t *Client) TemplateInstanceIsReady() (bool, error) {
+	ti, err := t.tc.TemplateInstances(t.namespace).Get(t.namespace, metav1.GetOptions{})
+	if kerrors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	for _, cond := range ti.Status.Conditions {
+		if cond.Type == templatev1.TemplateInstanceReady &&
+			cond.Status == corev1.ConditionTrue {
+			return true, nil
+		} else if cond.Type == templatev1.TemplateInstanceInstantiateFailure &&
+			cond.Status == corev1.ConditionTrue {
+			return false, fmt.Errorf("templateinstance %q failed", t.namespace)
+		}
+	}
+	return false, nil
+}
+
+func (t *Client) DeploymentConfigIsReady(name string, replicas int32) func() (bool, error) {
+	return func() (bool, error) {
+		dc, err := t.ac.DeploymentConfigs(t.namespace).Get(name, metav1.GetOptions{})
+		switch {
+		case err == nil:
+			return replicas == dc.Status.Replicas &&
+				replicas == dc.Status.ReadyReplicas &&
+				replicas == dc.Status.AvailableReplicas &&
+				replicas == dc.Status.UpdatedReplicas &&
+				dc.Generation == dc.Status.ObservedGeneration, nil
+		default:
+			return false, err
+		}
+	}
+}
+
+func (t *Client) ProjectIsCleanedUp() (bool, error) {
+	_, err := t.pc.Projects().Get(t.namespace, metav1.GetOptions{})
+	if kerrors.IsNotFound(err) {
+		return true, nil
+	}
+	return false, err
+}
+
+func (t *Client) DefaultServiceAccountIsReady() (bool, error) {
+	sa, err := t.kc.CoreV1().ServiceAccounts(t.namespace).Get("default", metav1.GetOptions{})
+	if kerrors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return len(sa.Secrets) > 0, nil
+}
+
+func (t *Client) SelfSarSuccess() (bool, error) {
+	res, err := t.kc.AuthorizationV1().SelfSubjectAccessReviews().Create(
+		&authorizationapiv1.SelfSubjectAccessReview{
+			Spec: authorizationapiv1.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationapiv1.ResourceAttributes{
+					Namespace: t.namespace,
+					Verb:      "create",
+					Resource:  "pods",
+				},
+			},
+		},
+	)
+	if err != nil {
+		return false, err
+	}
+	return res.Status.Allowed, nil
+}
+
+func (t *Client) CreatePodDisruptionBudget(p *policy.PodDisruptionBudget) error {
+	_, err := t.kc.Policy().PodDisruptionBudgets(t.namespace).Create(p)
+	return err
+}
+
+func (t *Client) GetRoute(template string, options *metav1.GetOptions) (*v1.Route, error) {
```

Comment:
Please inline back all wrapper function that wrap single-line api calls. Trust me, you don't want to maintain these going forward. We have been through the pain before.